SIGSEGV: segmentation violation on GCE SD without Labels when upgrading from 1.84.0 to 1.85.3

### Describe the bug

`vmagent` crashes during Service Discovery on GCE with vms without Labels.

It's probably caused by [this line](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/e640ff72f1336d09d3e53c8581e2d6ce5af32c92/lib/promscrape/discovery/gce/instance.go#L163): 
```go
	for _, label := range inst.Labels.Labels {
		m.Add(discoveryutils.SanitizeLabelName("__meta_gce_label_"+label.Name), label.Value)
	}
```

When `inst.Labels` is null. 

There is no panic when running on _1.84.0_.



### To Reproduce

Here is the job config, which works fine in 1.84.0:
```yaml
- job_name: vm-instances-on-gce
  scheme: https
  honor_labels: true
  gce_sd_configs:
    - zone: europe-central2-a
      project: traefiklabs
  tls_config:
    insecure_skip_verify: true
  basic_auth:
    username: prometheus
    password_file: /etc/gce-node-exporter/node-exporter-password
  relabel_configs:
    - source_labels: [__meta_gce_public_ip]
      target_label: __address__
      replacement: "$1:9100"
    - source_labels: [__meta_gce_zone]
      regex: ".+/([^/]+)"
      target_label: region
    - source_labels: [__meta_gce_project]
      target_label: project
    - source_labels: [__meta_gce_instance_name]
      target_label: instance
    - regex: "__meta_gce_metadata_(.+)"
      action: labelmap
  metric_relabel_configs:
    - action: replace
      source_labels:
        - instance
      target_label: node_name
    - action: keep
      regex: (node_network_receive_bytes_total|node_network_transmit_bytes_total|node_filesystem_avail_bytes|node_filesystem_size_bytes|node_disk_io_time_seconds_total|node_disk_written_bytes_total|node_disk_read_bytes_total|node_memory_MemAvailable_bytes|node_cpu_count|node_memory_MemTotal_bytes|node_memory_MemFree_bytes|node_memory_Buffers_bytes|node_memory_Cached_bytes|node_load1|node_load5|node_load15|node_cpu_seconds_total)
      source_labels: [__name__]
    - action: labeldrop
      regex: (project|startup_script)
```

### Version

$ docker run victoriametrics/vmagent:v1.85.3 --version
vmagent-20221220-230030-tags-v1.85.3-0-gf443fad56

### Logs

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x407a6c]

goroutine 85 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/gce.(*Instance).appendTargetLabels(0x40002dd9b8, {0x0, 0x0, 0x0}, {0x40001f4e50, 0xb}, {0x69f95d, 0x1}, 0x4a1e80?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/gce/instance.go:163 +0xafc
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/gce.getInstancesLabels(0x40001f0360)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/gce/instance.go:19 +0xe0
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/gce.(*SDConfig).GetLabels(0x40001fbb78?, {0x1bec4?, 0x0?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/discovery/gce/gce.go:70 +0x74
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.appendSDScrapeWork({0x9ae368, 0x0, 0x0}, {0x6a2fe0?, 0x40000b8480?}, {0xffffd74e5458?, 0x40001e2c60?}, 0x400006b110, {0x529e06, 0xd})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/config.go:1005 +0x54
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.(*Config).getGCESDScrapeWork(0x4000063e60, {0x0, 0x0, 0x438368?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/config.go:733 +0x350
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.runScraper.func10(0x40001b0b60?, {0x0?, 0x40001fbe38?, 0x98380?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/scraper.go:130 +0x20
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.(*scrapeConfig).run.func1(0xdf8475800?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/scraper.go:278 +0x74
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.(*scrapeConfig).run(0x4000151b00, 0x0?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/scraper.go:287 +0x124
github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.(*scrapeConfigs).add.func1()
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/scraper.go:235 +0x5c
created by github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape.(*scrapeConfigs).add
	github.com/VictoriaMetrics/VictoriaMetrics/lib/promscrape/scraper.go:233 +0x230


### Screenshots

_No response_

### Used command-line flags

_No response_

### Additional information

_No response_

--- a/docs/CHANGELOG.md
+++ b/docs/CHANGELOG.md
@@ -46,7 +46,7 @@ The following tip changes can be tested by building VictoriaMetrics components f
 * BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent.html): fix possible resource leak after hot reload of the updated [consul_sd_configs](https://docs.victoriametrics.com/sd_configs.html#consul_sd_configs). See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3468).
 * BUGFIX: properly return label names starting from uppercase such as `CamelCaseLabel` from [/api/v1/labels](https://docs.victoriametrics.com/url-examples.html#apiv1labels). See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3566).
 * BUGFIX: consistently select the sample with the biggest value out of samples with identical timestamps during querying when the [deduplication](https://docs.victoriametrics.com/#deduplication) is enabled according to [this feature request](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3333). Previously random samples could be selected during querying.
-
+* BUGFIX: [vmagent](https://docs.victoriametrics.com/vmagent.html): fix a panic during GCE service discovery for instance without any labels set. See [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/3624).
 
 ## [v1.85.3](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.85.3)
 

--- a/lib/promscrape/discovery/gce/instance.go
+++ b/lib/promscrape/discovery/gce/instance.go
@@ -160,8 +160,10 @@ func (inst *Instance) appendTargetLabels(ms []*promutils.Labels, project, tagSep
 	for _, item := range inst.Metadata.Items {
 		m.Add(discoveryutils.SanitizeLabelName("__meta_gce_metadata_"+item.Key), item.Value)
 	}
-	for _, label := range inst.Labels.Labels {
-		m.Add(discoveryutils.SanitizeLabelName("__meta_gce_label_"+label.Name), label.Value)
+	if inst.Labels != nil {
+		for _, label := range inst.Labels.Labels {
+			m.Add(discoveryutils.SanitizeLabelName("__meta_gce_label_"+label.Name), label.Value)
+		}
 	}
 	if len(iface.AccessConfigs) > 0 {
 		ac := iface.AccessConfigs[0]

